File: pyg_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbItemCat;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface ItemCatService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbItemCat> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    void add(TbItemCat itemCat);


    /**
     * 修改
     */
    void update(TbItemCat itemCat);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbItemCat findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPage(TbItemCat itemCat, int pageNum, int pageSize);

    List<TbItemCat> findItemCatByParentId(Long parentId);
}

File: pyg_shop_web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../itemCat/findAll');
    };
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../itemCat/findPage/' + page + '/' + rows);
    };
    //查询实体
    this.findOne = function (id) {
        return $http.get('../itemCat/findOne/' + id);
    };
    //增加
    this.add = function (entity) {
        return $http.post('../itemCat/add', entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post('../itemCat/update', entity);
    };
    //删除
    this.dele = function (ids) {
        return $http.get('../itemCat/delete/' + ids);
    };
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../itemCat/search/' + page + "/" + rows, searchEntity);
    };
    //根据parentId查询下级分类
    this.findItemCatByParentId = function (parentId) {
        return $http.post('../itemCat/findItemCatByParentId/' + parentId);
    }

});
